### ISSUE TITLE
Make Jbrowse CLI tools work reliably in Biocontainers

### ISSUE THREAD

**Issue by fubar2** [GMOD/jbrowse-components#4608]
tl;dr _JBrowse2 CL failing in a Galaxy tool._

Unable to implement PIF indexes - the Galaxy tool does not use the built in jbrowse xxx commands otherwise.

Added a call to make-pif and tests pass without problems if run using Conda, but fail when tested using the JBrowse2 Biocontainer:

```
DEBUG:jbrowse:cd /tmp/tmpzefftlib/job_working_directory/000/17/outputs/dataset_1e09fe24-141b-439a-ba4d-e72140449478_files && cp /tmp/tmpzefftlib/files/c/c/5/dataset_cc53e7f1-20f8-44df-9069-dd99d8485e0b.dat ./temppif
DEBUG:jbrowse:cd /tmp/tmpzefftlib/job_working_directory/000/17/outputs/dataset_1e09fe24-141b-439a-ba4d-e72140449478_files && jbrowse make-pif ./temppif --out ./peach-grape-map.paf_10.pif.gz
node:os:359
    throw new ERR_SYSTEM_ERROR(ctx);
          ^

SystemError [ERR_SYSTEM_ERROR]: A system error occurred: uv_os_get_passwd returned ENOENT (no such file or directory)
    at userInfo (node:os:359:11)
    at Config._shell (/usr/local/lib/node_modules/@jbrowse/cli/node_modules/@oclif/core/lib/config/config.js:587:67)
    at Config.load (/usr/local/lib/node_modules/@jbrowse/cli/node_modules/@oclif/core/lib/config/config.js:309:27)
    at async Config.load (/usr/local/lib/node_modules/@jbrowse/cli/node_modules/@oclif/core/lib/config/config.js:167:9)
    at async Object.run (/usr/local/lib/node_modules/@jbrowse/cli/node_modules/@oclif/core/lib/main.js:62:20) {
  code: 'ERR_SYSTEM_ERROR',
  info: {
    errno: -2,
    code: 'ENOENT',
    message: 'no such file or directory',
    syscall: 'uv_os_get_passwd'
  },
  errno: [Getter/Setter],
  syscall: [Getter/Setter]
}

Node.js v22.9.0
```
Suggestion from @garrettjstevens via @scottcain - 
```Reasoning: the CLI framework we use probably calls a nodejs api that throws this error if the user running it has no username or homedir (see https://nodejs.org/docs/latest-v20.x/api/os.html#osuserinfooptions)```

**Describe the solution you'd like**

Running the container as nobody - jbrowse make-pif seems to work.

The biocontainer is automatically built from a simple recipe containing all the Conda dependency identifiers that need to be added in: https://github.com/BioContainers/multi-package-containers/blob/8d9e68e37afc2d8a546bef110a52d8e8d8a825e7/combinations/mulled-v2-7c1118c363d59676a7626322b826de7dc8da5c1a%3A6d5de92e647715f65170e63dc4db9b4a02c826b4-0.tsv#L2

It can be downloaded from the repository for testing if that helps?

`docker pull quay.io/biocontainers/jbrowse2:2.15.4--h5284365_0`

**Describe alternatives you've considered**

Avoid all JBrowse2 command line tools. 

That works so far and the tool successfully generates JSON directly for configuration files and default sessions, rather than using the CLI.  Everything is working really well at present without them but PIF is impossible to implement in the Galaxy JBrowse2 tool without the CLI as there is no alternative indexer. Working with some huge VGP PAF tracks which are frustratingly slow, particularly as dotplots, without a tabix style index. Reverting to using PAF :( 

**Additional context**
This makes the CL tools unusable in any Biocontainer. Many large scale Galaxy services use Biocontainers for security and reproducibility so all tools must be reliable when running in that context.

**Comment by scottcain** [comment]
I guess one fix would be to write a tool in some other language (python, perl) that can generate a pif. @cmdcolin any idea how hard that would be?

**Comment by fubar2** [comment]
Temporarily closing this - everything works in the biocontainer so this may be a Galaxy problem - will investigate further.

**Comment by fubar2** [comment]
update: @cmdcolin - turned out this whole saga was a Biocontainers/node.js interaction so not JBrowse2 codebase problem, and easily fixed in the Galaxy tool code. 

No idea if this will help anyone else but adding `SHELL="/bin/sh"` to the env when calling subprocess.run in python, allows the jbrowse make-pif command to run normally.  node-js's honour appears to be satisfied....
